tests/ClientBuilderTest.php

```diff
@@ -210,6 +210,33 @@ public function optionsDataProvider()
             ['setErrorTypes', 0],
         ];
     }
+
+    /**
+     * @dataProvider compressionOptionDataProvider
+     */
+    public function testCompressionPlugin($enabled)
+    {
+        $builder = ClientBuilder::create(['enable_compression' => $enabled, 'dsn' => 'http://public:secret@example.com/sentry/1']);
+        $builder->getClient();
+
+        $decoderPluginFound = false;
+        foreach ($this->getObjectAttribute($builder, 'httpClientPlugins') as $plugin) {
+            if ($plugin instanceof Plugin\DecoderPlugin) {
+                $decoderPluginFound = true;
+                break;
+            }
+        }
+
+        $this->assertEquals($enabled, $decoderPluginFound);
+    }
+
+    public function compressionOptionDataProvider()
```

Comment:
Typehint the return type (`array`) and change the method name to `getClientTogglesCompressionPluginInHttpClientDataProvider`